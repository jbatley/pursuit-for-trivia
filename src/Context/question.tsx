import { createContext } from 'react';

const QuestionContext = createContext({
  a: () => {},
});

export default QuestionContext;
