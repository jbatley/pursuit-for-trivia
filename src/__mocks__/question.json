{
  "response_code": 0,
  "results": [
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggY29tcGFueSBkZXZlbG9wZWQgdGhlIHZpZGVvIGdhbWUgIkJvcmRlcmxhbmRzIj8=",
      "correct_answer": "R2VhcmJveCBTb2Z0d2FyZQ==",
      "incorrect_answers": [
        "MksgR2FtZXM=",
        "QWN0aXZpc2lvbg==",
        "Um9ja3N0YXIgR2FtZXM="
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hlbiB3YXMgdGhlIERlY2xhcmF0aW9uIG9mIEluZGVwZW5kZW5jZSBhcHByb3ZlZCBieSB0aGUgU2Vjb25kIENvbnRpbmVudGFsIENvbmdyZXNzPw==",
      "correct_answer": "SnVseSAyLCAxNzc2",
      "incorrect_answers": [
        "TWF5IDQsIDE3NzY=",
        "SnVuZSA0LCAxNzc2",
        "SnVseSA0LCAxNzc2"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hvIHdhcyB0aGUgZmlyc3QgamVkaSB0aGF0IFN0YXJraWxsZXIgaGFkIHRvIGtpbGwgaW4gU3RhciBXYXJzOiBUaGUgRm9yY2UgVW5sZWFzaGVkPw==",
      "correct_answer": "UmFobSBLb3Rh",
      "incorrect_answers": [
        "QWhzb2thIFRhbm8=",
        "S2F6ZGFuIFBhcmF0dXM=",
        "S2VudG8gTWFyZWs="
      ]
    },
    {
      "category": "SGlzdG9yeQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "IFdoYXQgUnVzc2lhbiBhdXRvbWF0aWMgZ2FzLW9wZXJhdGVkIGFzc2F1bHQgcmlmbGUgd2FzIGRldmVsb3BlZCBpbiB0aGUgU292aWV0IFVuaW9uIGluIDE5NDcsIGFuZCBpcyBzdGlsbCBwb3B1bGFybHkgdXNlZCB0b2RheT8=",
      "correct_answer": "QUstNDc=",
      "incorrect_answers": [
        "UlBL",
        "TTE2",
        "TUcgNDI="
      ]
    },
    {
      "category": "R2VvZ3JhcGh5",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SG93IG1hbnkgY291bnRyaWVzIGFyZSBpbnNpZGUgdGhlIFVuaXRlZCBLaW5nZG9tPw==",
      "correct_answer": "Rm91cg==",
      "incorrect_answers": [
        "VHdv",
        "VGhyZWU=",
        "T25l"
      ]
    },
    {
      "category": "U3BvcnRz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "QSBzdGltcG1ldGVyIG1lYXN1cmVzIHRoZSBzcGVlZCBvZiBhIGJhbGwgb3ZlciB3aGF0IHN1cmZhY2U\/",
      "correct_answer": "R29sZiBQdXR0aW5nIEdyZWVu",
      "incorrect_answers": [
        "IEZvb3RiYWxsIFBpdGNo",
        "Q3JpY2tldCBPdXRmaWVsZA==",
        "UGluYmFsbCBUYWJsZQ=="
      ]
    },
    {
      "category": "VmVoaWNsZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggY291bnRyeSBoYXMgdGhlIGludGVybmF0aW9uYWwgdmVoaWNsZSByZWdpc3RyYXRpb24gbGV0dGVyICdBJz8=",
      "correct_answer": "QXVzdHJpYQ==",
      "incorrect_answers": [
        "QWZnaGFuaXN0YW4=",
        "QXVzdHJhbGlh",
        "QXJtZW5pYQ=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogSmFwYW5lc2UgQW5pbWUgJiBNYW5nYQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gdGhlIGFuaW1lICJNci4gT3NvbWF0c3UiLCBob3cgbWFueSBicm90aGVycyBkb2VzIE9zb21hdHN1LXNhbiBoYXZlPw==",
      "correct_answer": "NQ==",
      "incorrect_answers": [
        "Ng==",
        "Nw==",
        "NA=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogQm9va3M=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCB3YXMgdGhlIHBlbiBuYW1lIG9mIG5vdmVsaXN0LCBNYXJ5IEFubiBFdmFucz8=",
      "correct_answer": "R2VvcmdlIEVsaW90",
      "incorrect_answers": [
        "R2VvcmdlIE9yd2VsbA==",
        "R2VvcmdlIEJlcm5hcmQgU2hhdw==",
        "R2VvcmdlIFNhdW5kZXJz"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogQm9hcmQgR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "QXQgdGhlIHN0YXJ0IG9mIGEgc3RhbmRhcmQgZ2FtZSBvZiB0aGUgTW9ub3BvbHksIGlmIHlvdSB0aHJvdyBhIGRvdWJsZSBzaXgsIHdoaWNoIHNxdWFyZSB3b3VsZCB5b3UgbGFuZCBvbj8=",
      "correct_answer": "RWxlY3RyaWMgQ29tcGFueQ==",
      "incorrect_answers": [
        "V2F0ZXIgV29ya3M=",
        "Q2hhbmNl",
        "Q29tbXVuaXR5IENoZXN0"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlc2UgZmVhdHVyZXMgd2FzIGFkZGVkIGluIHRoZSAxOTk0IGdhbWUgIkhlcmV0aWMiIHRoYXQgdGhlIG9yaWdpbmFsICJET09NIiBjb3VsZCBub3QgYWRkIGR1ZSB0byBsaW1pdGF0aW9ucz8=",
      "correct_answer": "TG9va2luZyB1cCBhbmQgZG93bg==",
      "incorrect_answers": [
        "SW5jcmVhc2VkIHJvb20gc2l6ZXM=",
        "VW5saW1pdGVkIHdlYXBvbnM=",
        "SGlnaGx5LWRldGFpbGVkIHRleHR1cmVz"
      ]
    },
    {
      "category": "VmVoaWNsZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlIGZvbGxvd2luZyB2ZWhpY2xlcyBmZWF0dXJlZCBhIGZ1bGwgZ2xhc3Mgcm9vZiBhdCBiYXNlIG1vZGVsPw==",
      "correct_answer": "UmVuYXVsdCBBdmFudGltZQ==",
      "incorrect_answers": [
        "Q2hldnkgVm9sdA==",
        "TWVyY2VkZXMtQmVueiBBLUNsYXNz",
        "SG9uZGEgT2R5c3NleQ=="
      ]
    },
    {
      "category": "U2NpZW5jZTogQ29tcHV0ZXJz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCB3YXMgdGhlIGZpcnN0IGNvbW1lcmljYWxseSBhdmFpbGFibGUgY29tcHV0ZXIgcHJvY2Vzc29yPw==",
      "correct_answer": "SW50ZWwgNDAwNA==",
      "incorrect_answers": [
        "SW50ZWwgNDg2U1g=",
        "VE1TIDEwMDA=",
        "QU1EIEFNMzg2"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogTXVzaWNhbHMgJiBUaGVhdHJlcw==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlc2UgbXVzaWNhbHMgd29uIHRoZSBUb255IEF3YXJkIGZvciBCZXN0IE11c2ljYWw\/",
      "correct_answer": "UmVudA==",
      "incorrect_answers": [
        "VGhlIENvbG9yIFB1cnBsZQ==",
        "QW1lcmljYW4gSWRpb3Q=",
        "TmV3c2llcw=="
      ]
    },
    {
      "category": "SGlzdG9yeQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hlbiB3YXMgdGhlIFVuaXRlZCBTdGF0ZXMgTmF0aW9uYWwgU2VjdXJpdHkgQWdlbmN5IGVzdGFibGlzaGVkPw==",
      "correct_answer": "Tm92ZW1iZXIgNCwgMTk1Mg==",
      "incorrect_answers": [
        "SnVseSAyNiwgMTkwOA==",
        "IEp1bHkgMSwgMTk3Mw==",
        "IE5vdmVtYmVyIDI1LCAyMDAy"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogTXVzaWM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCdzIHRoZSBtb3N0IGNvbW1vbiB0aW1lIHNpZ25hdHVyZSBmb3Igcm9jayBzb25ncz8=",
      "correct_answer": "NC80",
      "incorrect_answers": [
        "MS8y",
        "OC8xMg==",
        "Mi80"
      ]
    },
    {
      "category": "U2NpZW5jZSAmIE5hdHVyZQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBwb2x5bWVyIGlzIHVzZWQgdG8gbWFrZSBDRHMsIHNhZmV0eSBnb2dnbGVzIGFuZCByaW90IHNoaWVsZHM\/",
      "correct_answer": "UG9seWNhcmJvbmF0ZQ==",
      "incorrect_answers": [
        "UnViYmVy",
        "Tnlsb24=",
        "QmFrZWxpdGU="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogTXVzaWM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlc2UgaXMgdGhlIG5hbWUgb2YgYW4gQW1lcmljYW4gcHN5Y2hlZGVsaWMgcm9jayBiYW5kIGZvcm1lZCBpbiAyMDAyIGJ5IEJlbmphbWluIEdvbGR3YXNzZXIgYW5kIEFuZHJldyBWYW5XeW5nYXJkZW4\/",
      "correct_answer": "TUdNVA==",
      "incorrect_answers": [
        "TVNUUktSRlQ=",
        "U1RSRktS",
        "U0JUUktU"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogQ29taWNz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgbmFtZSBvZiB0aGUgbWFpbiBjaGFyYWN0ZXIgaW4gdGhlIHdlYmNvbWljIEd1bm5lcmtyaWdnIENvdXJ0IGJ5IFRvbSBTaWRkZWxsPw==",
      "correct_answer": "QW50aW1vbnk=",
      "incorrect_answers": [
        "QmlzbXV0aA==",
        "TWVyY3VyeQ==",
        "Q29iYWx0"
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgU3dlZGlzaCB3b3JkIGZvciAid2luZG93Ij8=",
      "correct_answer": "RsO2bnN0ZXI=",
      "incorrect_answers": [
        "SMOlbA==",
        "U2vDpHJt",
        "UnV0YQ=="
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgSXRhbGlhbiB3b3JkIGZvciAidG9tYXRvIj8=",
      "correct_answer": "UG9tb2Rvcm8=",
      "incorrect_answers": [
        "QWdsaW8=",
        "Q2lwb2xsYQ==",
        "UGVwZXJvbmNpbm8="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "Q2FwY29tJ3Mgc3Vydml2YWwgaG9ycm9yIHRpdGxlIERlYWQgUmlzaW5nLCBjYW5vbmljYWxseSBzdGFydHMgb24gd2hhdCBkYXkgb2YgU2VwdGVtYmVyIDIwMDY\/",
      "correct_answer": "U2VwdGVtYmVyIDE5dGg=",
      "incorrect_answers": [
        "U2VwdGVtYmVyIDIxc3Q=",
        "U2VwdGVtYmVyIDMwdGg=",
        "U2VwdGVtYmVyIDE0dGg="
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgbGFzdCBsZXR0ZXIgb2YgdGhlIEdyZWVrIGFscGhhYmV0Pw==",
      "correct_answer": "T21lZ2E=",
      "incorrect_answers": [
        "TXU=",
        "RXBzaWxvbg==",
        "S2FwcGE="
      ]
    },
    {
      "category": "U2NpZW5jZSAmIE5hdHVyZQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBkb2VzIHRoZSB5ZWxsb3cgZGlhbW9uZCBvbiB0aGUgTkZQQSA3MDQgZmlyZSBkaWFtb25kIHJlcHJlc2VudD8=",
      "correct_answer": "UmVhY3Rpdml0eQ==",
      "incorrect_answers": [
        "SGVhbHRo",
        "RmxhbW1hYmlsaXR5",
        "UmFkaW9hY3Rpdml0eQ=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hvIGNyZWF0ZWQgdGhlIHB1bXAgIkYuTC5VLkQuRC4iIE1hcmlvIHVzZXMgaW4gU3VwZXIgTWFyaW8gU3Vuc2hpbmU\/",
      "correct_answer": "RWx2aW4gR2FkZA==",
      "incorrect_answers": [
        "Um9iZXJ0IEZsdWRk",
        "Tmlyb25h",
        "Q3J5Z29y"
      ]
    },
    {
      "category": "SGlzdG9yeQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCB3YXMgdGhlIHRyYW5zZmVyIG9mIGRpc2Vhc2UsIGNyb3BzLCBhbmQgcGVvcGxlIGFjcm9zcyB0aGUgQXRsYW50aWMgc2hvcnRseSBhZnRlciB0aGUgZGlzY292ZXJ5IG9mIHRoZSBBbWVyaWNhcyBjYWxsZWQ\/",
      "correct_answer": "VGhlIENvbHVtYmlhbiBFeGNoYW5nZQ==",
      "incorrect_answers": [
        "VHJpYW5nbGUgVHJhZGU=",
        "VHJhbnNhdGxhbnRpYyBTbGF2ZSBUcmFkZQ==",
        "VGhlIFNpbGsgUm9hZA=="
      ]
    },
    {
      "category": "U2NpZW5jZSAmIE5hdHVyZQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "QXQgd2hhdCB0ZW1wZXJhdHVyZSBkb2VzIHdhdGVyIGJvaWw\/",
      "correct_answer": "MjEywrBG",
      "incorrect_answers": [
        "MjAwwrBG",
        "MTgxwrBG",
        "MTc4wrBG"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVGVsZXZpc2lvbg==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SG93IGxvbmcgd2FzIEtlbiBKZW5uaW5ncycgd2luIHN0cmVhayBvbiBKZW9wYXJkeT8=",
      "correct_answer": "NzQ=",
      "incorrect_answers": [
        "ODg=",
        "NDk=",
        "NjI="
      ]
    },
    {
      "category": "U2NpZW5jZTogQ29tcHV0ZXJz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "LmF0IGlzIHRoZSB0b3AtbGV2ZWwgZG9tYWluIGZvciB3aGF0IGNvdW50cnk\/",
      "correct_answer": "QXVzdHJpYQ==",
      "incorrect_answers": [
        "QXJnZW50aW5h",
        "QXVzdHJhbGlh",
        "QW5nb2xh"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogQm9hcmQgR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gQ2hlc3MsIHRoZSBRdWVlbiBoYXMgdGhlIGNvbWJpbmVkIG1vdmVtZW50IG9mIHdoaWNoIHR3byBwaWVjZXM\/",
      "correct_answer": "QmlzaG9wIGFuZCBSb29r",
      "incorrect_answers": [
        "Um9vayBhbmQgS2luZw==",
        "S25pZ2h0IGFuZCBCaXNob3A=",
        "S2luZyBhbmQgS25pZ2h0"
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogRmlsbQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hvIHZvaWNlZCBNZXRhbGJlYXJkIGluICJUaGUgTGVnbyBNb3ZpZSI\/",
      "correct_answer": "TmljayBPZmZlcm1hbg==",
      "incorrect_answers": [
        "TGlhbSBOZWVzb24=",
        "TW9yZ2FuIEZyZWVtYW4=",
        "V2lsbCBBcm5ldA=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogTXVzaWM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlIGZvbGxvd2luZyBiYW5kcyBpcyBUb20gRGVMb25nZSBub3QgYSBwYXJ0IG9mPw==",
      "correct_answer": "KzQ0",
      "incorrect_answers": [
        "Qm94IENhciBSYWNlcg==",
        "QmxpbmstMTgy",
        "QW5nZWxzICYgQWlyd2F2ZXM="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBlbGVtZW50IGRvZXMgdGhlIFppbm9ncmUgdXNlIGluIE1vbnN0ZXIgSHVudGVyPw==",
      "correct_answer": "VGh1bmRlcg==",
      "incorrect_answers": [
        "SWNl",
        "RmlyZQ==",
        "V2F0ZXI="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVGVsZXZpc2lvbg==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggQnJpdGlzaCB3cml0ZXIgd3JvdGUgZm9yIGJvdGggRG9jdG9yIFdobyBhbmQgU2hlcmxvY2s\/",
      "correct_answer": "U3RldmVuIE1vZmZhdHQ=",
      "incorrect_answers": [
        "UGhpbCBGb3Jk",
        "UnVzc2VsbCBUIERhdmllcw==",
        "VG9ieSBXaGl0aG91c2U="
      ]
    },
    {
      "category": "SGlzdG9yeQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hvIHdhcyB0aGUgbGVhZGVyIG9mIHRoZSBDb21tdW5pc3QgUGFydHkgb2YgWXVnb3NsYXZpYSA\/",
      "correct_answer": "Sm9zaXAgQnJveiBUaXRv",
      "incorrect_answers": [
        "S2FyYWRqb3JkamUgUGV0cm92aWM=",
        "TWlsb3MgT2JpbGlj",
        "QWxlc2thbmRhciBQZXRyb3ZpYw=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gIkZhbGxvdXQgNCIsIHdoYXQgaXMgdGhlIG5hbWUgb2YgdGhlIGRvZyB5b3UgZmluZCBhdCBSZWQgUm9ja2V0IHRydWNrIHN0b3A\/",
      "correct_answer": "RG9nbWVhdA==",
      "incorrect_answers": [
        "U3Bhcmt5",
        "UXVhemFy",
        "Q2hvcA=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogQm9hcmQgR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlIGZvbGxvd2luZyB0YWJsZXRvcCBnYW1lcyBpcyB0aGUgb2xkZXN0Pw==",
      "correct_answer": "R28=",
      "incorrect_answers": [
        "Q2hlc3M=",
        "TWFoam9uZw==",
        "U2hvZ2k="
      ]
    },
    {
      "category": "U2NpZW5jZTogQ29tcHV0ZXJz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlc2UgcGVvcGxlIHdhcyBOT1QgYSBmb3VuZGVyIG9mIEFwcGxlIEluYz8=",
      "correct_answer": "Sm9uYXRoYW4gSXZl",
      "incorrect_answers": [
        "U3RldmUgSm9icw==",
        "Um9uYWxkIFdheW5l",
        "U3RldmUgV296bmlhaw=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogSmFwYW5lc2UgQW5pbWUgJiBNYW5nYQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCB5ZWFyIGRpZCAiQmlzaG91am8gU2Vuc2hpIFNhaWxvciBNb29uIiBhaXIgaW4gSmFwYW4\/",
      "correct_answer": "MTk5Mg==",
      "incorrect_answers": [
        "MTk4OQ==",
        "MTk5MA==",
        "MTk5NA=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gIkNhbGwgT2YgRHV0eTogWm9tYmllcyIsIGNvbXBsZXRpbmcgd2hpY2ggbWFwJ3MgZWFzdGVyIGVnZyB3aWxsIHJld2FyZCB5b3Ugd2l0aCB0aGUgYWNoaWV2ZW1lbnQsICJUaW1lIFRyYXZlbCBXaWxsIFRlbGwiPw==",
      "correct_answer": "U2hhbmdyaS1MYQ==",
      "incorrect_answers": [
        "QXNjZW5zaW9u",
        "TW9vbg==",
        "RGllIFJpc2U="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogSmFwYW5lc2UgQW5pbWUgJiBNYW5nYQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gdGhlICJUbyBMb3ZlLVJ1IiBzZXJpZXMsIGhvdyBtYW55IFRyYW5zLXdlYXBvbnMgd2VyZSBjcmVhdGVkPw==",
      "correct_answer": "Mw==",
      "incorrect_answers": [
        "MQ==",
        "Mg==",
        "NA=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogRmlsbQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gIkp1cmFzc2ljIFdvcmxkIiwgd2hpY2ggY29tcGFueSBwdXJjaGFzZXMgSW5HZW4gYW5kIGNyZWF0ZXMgSnVyYXNzaWMgV29ybGQ\/",
      "correct_answer": "TWFzcmFuaSBHbG9iYWwgQ29ycG9yYXRpb24g",
      "incorrect_answers": [
        "QmlvbG9neSBTeW50aGV0aWNzIFRlY2hub2xvZ2llcw==",
        "SW50ZXJuYXRpb25hbCBHZW5ldGljcyBJbmNvcnBvcmF0ZWQ=",
        "SW50ZXJuYXRpb25hbCBHZW5ldGljIFRlY2hub2xvZ2llcw=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "SW4gIlJlc2lkZW50IEV2aWwgMiIsIHdoYXQgaXMgTGVvbiBLZW5uZWR5J3MgbWlkZGxlIG5hbWU\/",
      "correct_answer": "U2NvdHQ=",
      "incorrect_answers": [
        "U2hhdW4=",
        "U2ltb24=",
        "U2Ft"
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggaXRlbSBvZiBjbG90aGluZyBpcyB1c3VhbGx5IHdvcm4gYnkgYSBTY290c21hbiBhdCBhIHdlZGRpbmc\/",
      "correct_answer": "S2lsdA==",
      "incorrect_answers": [
        "U2tpcnQ=",
        "RHJlc3M=",
        "UmhvYmVz"
      ]
    },
    {
      "category": "R2VuZXJhbCBLbm93bGVkZ2U=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlc2Ugd29yZHMgbWVhbnMgImlkbGUgc3BlY3RhdG9yIj8=",
      "correct_answer": "R29uZ29vemxlcg==",
      "incorrect_answers": [
        "R29zc3lwaWJvbWE=",
        "SmVudGFjdWxhcg==",
        "TWV1cGFyZXVuaWE="
      ]
    },
    {
      "category": "U3BvcnRz",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggc3BvcnQgaXMgTk9UIHRyYWRpdGlvbmFsbHkgcGxheWVkIGR1cmluZyB0aGUgTW9uZ29saWFuIE5hYWRhbSBmZXN0aXZhbD8=",
      "correct_answer": "QW1lcmljYW4gRm9vdGJhbGw=",
      "incorrect_answers": [
        "V3Jlc3RsaW5n",
        "QXJjaGVyeQ==",
        "SG9yc2UtUmFjaW5n"
      ]
    },
    {
      "category": "U2NpZW5jZSAmIE5hdHVyZQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgaGFsZi1saWZlIG9mIFVyYW5pdW0tMjM1Pw==",
      "correct_answer": "NzAzLDgwMCwwMDAgeWVhcnM=",
      "incorrect_answers": [
        "NCwzMDAsNDAwLDAwMCB5ZWFycw==",
        "MSwyNjAsOTAwLDAwMCB5ZWFycw==",
        "VXJhbml1bS0yMzUgaXMgYSBzdGFibGUgaXNvdG9wZQ=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogRmlsbQ==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggc3RhbmQtdXAgY29tZWRpYW4gdm9pY2VkIHRoZSB0YWxraW5nIHBhcnJvdCAiSWFnbyIgaW4gRGlzbmV5J3MgMTk5MiBhZGFwdGF0aW9uIG9mIEFsYWRkaW4\/",
      "correct_answer": "R2lsYmVydCBHb3R0ZnJpZWQ=",
      "incorrect_answers": [
        "Um9iaW4gV2lsbGlhbXM=",
        "UGF1bHkgU2hvcmU=",
        "Sm9uYXRoYW4gRnJlZW1hbg=="
      ]
    },
    {
      "category": "RW50ZXJ0YWlubWVudDogVmlkZW8gR2FtZXM=",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hpY2ggb2YgdGhlIGZvbGxvd2luZyBNYXJpbyBLYXJ0IDggRGVsdXhlIGl0ZW1zIHdpbGwgTk9UIG1ha2UgeW91IGludmluY2libGU\/",
      "correct_answer": "R29sZGVuIE11c2hyb29t",
      "incorrect_answers": [
        "U3Rhcg==",
        "QnVsbGV0IEJpbGw=",
        "Qm9v"
      ]
    },
    {
      "category": "QW5pbWFscw==",
      "type": "bXVsdGlwbGU=",
      "difficulty": "bWVkaXVt",
      "question": "V2hhdCBpcyB0aGUgbmFtZSBmb3IgYSBtYWxlIGJlZSB0aGF0IGNvbWVzIGZyb20gYW4gdW5mZXJ0aWxpemVkIGVnZz8=",
      "correct_answer": "RHJvbmU=",
      "incorrect_answers": [
        "U29sZGllcg==",
        "V29ya2Vy",
        "TWFsZQ=="
      ]
    }
  ]
}
