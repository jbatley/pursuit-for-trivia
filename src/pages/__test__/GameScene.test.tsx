import React from 'react';
import { render as testRender } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import ProviderWrapper from 'ProvidersWrapper';
import question from '__mocks__/question.json';
import Component from '../GameScene';

function render(children: React.ReactChild) {
  return testRender(
    <ProviderWrapper>
      <Router>
        {children}
      </Router>
    </ProviderWrapper>,
  );
}

describe('Page GameScene', () => {
  test('should render and', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify(question),
      { status: 200 },
    );
    const { container } = render(<Component />);
    expect(container.firstChild).not.toBe(null);
  });
  test('matches snapshot', async () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
