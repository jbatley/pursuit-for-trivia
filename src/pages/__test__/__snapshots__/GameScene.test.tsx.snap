// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page GameScene matches snapshot 1`] = `
<canvas
  class="canvas"
/>
`;
